src/comm/TcpAcceptor.cc

```diff
@@ -426,7 +426,8 @@ Comm::TcpAcceptor::oldAccept(Comm::ConnectionPointer &details)
     // set socket flags
     commSetCloseOnExec(sock);
     commSetNonBlocking(sock);
-    Comm::ApplyTcpKeepAlive(sock, listenPort_->tcp_keepalive);
+    if (listenPort_)
+        Comm::ApplyTcpKeepAlive(sock, listenPort_->tcp_keepalive);
```

Comment:
> Amos: FYI: this re-opens the issues with slow TCP connections being closed early due to lack of TCP keep-alive signals.

FYI: This particular PR/change does not reopen any issues. It has no effect on "slow TCP connections".

Please see Eduard's email on squid-bugs and discuss there if needed.

> I am not marking this for merge so @rousskov can decide how that is to be handled.

I am clearing this important bug fix for merging.